scripts/test_images.sh

```diff
@@ -82,13 +82,3 @@ if [ "${GET}" != "${VALUE}" ]; then
 fi
 
 echo "Succesfully tested etcd local image ${TAG}"
-
-for TARGET_ARCH in "amd64" "arm64" "ppc64le" "s390x"; do
```

Comment:
Can you update the testing of ARCH, so we validate that multi-arch image correctly links to subimages for all supported architectures?

Reply:
Sure thing - I noticed while I was working on it there is duplication of key variables and functions between `test_images.sh` and `release.sh` so I have refactored into a new `docker_lib.sh`, added a lot more error handling and improved consistency with how errors are reported. Let me know what you think.

Reply:
Actually there is a hitch here as well, when in dry run the images don't actually exist in the remote registry. With `docker manifest` only working on an actual registry rather than local, refer https://github.com/docker/cli/issues/3350

Reading through there are some potential hacks to work around, or we could spin up a temporary registry as part of ci to test this more naturally, not sure what the appetite for either is though.